sarpy/io/phase_history/cphd.py

```diff
@@ -408,7 +372,7 @@ def _read_pvp_vector(self, pvp_block_offset, pvp_offset, vector_size, field_offs
         with open(self.cphd_details.file_name, 'rb') as fi:
             for i in range(out_count):
                 fi.seek(current_offset)
-                element = struct.unpack('>{}{}'.format(siz, frm), fi.read(fld_siz))
+                element = numpy.fromfile(fi, dtype=dtype.newbyteorder('B'), count=siz)
```

Comment:
How does this account for the fact that the data is written as big-endian? That's why I was using struct, as I recall. I confess that it's been awhile since I crafted this. I'm sorry that you were wandering through all this, I had no idea.

Reply:
Definitely no need to apologize; it makes it much easier to have such a developed codebase to add to. The major reason to change away from `struct` here was the fact that `struct` and `numpy` have slightly different type characters (e.g. L = uint32 in struct but uint64 in numpy).
The [`newbyteorder('B')`](https://numpy.org/doc/stable/reference/generated/numpy.dtype.newbyteorder.html)  should enforce the byte order of the returned array.

Reply:
I was more worried about recognizing the original is explicitly stated as big-endian. It's probably wrapped into the dtype definition, but I missed it.